src/Compilers/Core/Portable/Operations/OperationInterfaces.xml

```diff
@@ -2899,4 +2899,21 @@
       </Comments>
     </Property>
   </Node>
+  <Node Name="IUsingVariableDeclarationOperation" Base="IOperation">
```

Comment:
>IUsingVariableDeclarationOperation [](start = 14, length = 34)

It looks like syntactically a Using Declaration is just a Local Declaration with additional  keywords. Should we simply reuse the existing node, IVariableDeclarationOperation, which already says that it is used to represent "(3) C# using declarations"?

Reply:
`IVariableDeclarationGroupOperation` is used as a child node of the regular using statement. So you have a parent `IUsingOperation` with a VariableDeclarationGroup and a body. That's the same for the other types mentioned in the comments.

This way we keep the same shape for using declarations: you have a parent using declaration, which contains a child declaration group of the variables declared. 

I originally went the route of updating DeclarationGroup with an enum on it to specify if it was using/async using/none, but this proved simpler as it was never used in any of the other cases, and this way you can keep the tree shape the same and visit the using declaration essentially the same as the statement.

Reply:
I am not following why would we need to make any changes to DeclarationGroup. Or why similarities with IUsingOperation are important. I think we should simply add IsUsing and IsAsynchronous properties to IVariableDeclarationOperation instead of adding new node. 

---
In reply to: [332695447](https://github.com/dotnet/roslyn/pull/39125#discussion_r332695447) [](ancestors = 332695447)

Reply:
I agree with @AlekseyTs here. That's how i would have expected thsi would be represented.

Reply:
Oh I see what you're referring to, that was from the initial 'dummy' implementation we did. Yep, we could definitely do that if you think it's better than introducing a new node?

Reply:
I agree with reusing the `IVariableDeclarationOperation` for this, though I'm not sure I agree with the two bool flags. These aren't independent variables: if `IsAsync` were true, `IsUsing` _must_, by definition, be true. I would likely prefer an enum over two bools. This will also give us flexibility if we add new types of variable declarations in the future, as we can add new kinds rather than throwing a whole bunch of flag booleans on the type itself.

Reply:
Yeah, enum is how I originally designed it interestingly enough, then split it out into its own operation, because it felt weird having most of the declarations have `UsingType=None`

Reply:
I wouldn't call it `UsingType`. I'd call it it `DeclarationType`, with options of `Standard`, `Using`, and `AsyncUsing`, or something along those lines.